SimulationAgent.Test/SimulationManagerTest.cs

```diff
@@ -254,6 +258,52 @@ void ItDeletesAllActorsAtTearDown()
             Assert.Empty(this.devicePropertiesActors);
         }
 
+        [Fact]
+        void ItCalculatesStatsAndInvokesCreateOrUpdateStatistics()
+        {
+            // Arrange
+            var expectedTotalMessageCount = 200;
+            var expectedFailedMessagesCount = 2;
+            var expectedFailedDeviceConnectionsCount = 6;
+            var expectedFailedTwinUpdatesCount = 10;
+
+            var statisticsModel = new SimulationStatisticsModel
+            {
+                TotalMessagesSent = expectedTotalMessageCount,
+                FailedMessagesCount = expectedFailedMessagesCount,
+                FailedDeviceConnectionsCount = expectedFailedDeviceConnectionsCount,
+                FailedDeviceTwinUpdatesCount = expectedFailedTwinUpdatesCount
+            };
+
+            var mockDeviceConnectionActor = new Mock<IDeviceConnectionActor>();
+            var mockDevicePropertiesActor = new Mock<IDevicePropertiesActor>();
+            var mockDeviceTelemetryActor = new Mock<IDeviceTelemetryActor>();
+
+            for (int i = 0; i < 2; i++)
+            {
+                var deviceName = SIM_ID + $"_{i}";
```

Comment:
use one or the other way of merging strings?  either `$"{SIM_ID}_{i}";` or `SIM_ID + "_" + i`